localstack/services/events/provider_v2.py

```diff
@@ -578,25 +635,36 @@ def _delete_target_sender(self, ids: TargetIdList, rule) -> None:
                 except KeyError:
                     LOG.error(f"Error deleting target service {target_arn}.")
 
-    def _put_entries(self, context: RequestContext, entries: PutEventsRequestEntryList) -> list:
+    def _put_entries(
+        self, context: RequestContext, entries: PutEventsRequestEntryList
+    ) -> tuple[list, list]:
+        entries_uuids = []
         failed_entries = []
         for event in entries:
             event_bus_name = event.get("EventBusName", "default")
+            event = format_event(event, context.region, context.account_id)
+            entries_uuids.append(event["id"])
             store = self.get_store(context)
-            event_bus = self.get_event_bus(event_bus_name, store)
-            # TODO add pattern matching
+            try:
+                event_bus = self.get_event_bus(event_bus_name, store)
+            except ResourceNotFoundException:
```

Comment:
this is validated AWS behavior - see tests `test_put_events_nonexistent_event_bus`